core/src/main/java/org/apache/iceberg/rest/requests/UpdateRequirementParser.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.requests.UpdateTableRequest.UpdateRequirement;
+import org.apache.iceberg.util.JsonUtil;
+
+public class UpdateRequirementParser {
+
+  private UpdateRequirementParser() {
+  }
+
+  private static final String TYPE = "type";
+
+  // assertion types
+  static final String ASSERT_TABLE_UUID = "assert-table-uuid";
+  static final String ASSERT_TABLE_DOES_NOT_EXIST = "assert-create";
+  static final String ASSERT_REF_SNAPSHOT_ID = "assert-ref-snapshot-id";
+  static final String ASSERT_LAST_ASSIGNED_FIELD_ID = "assert-last-assigned-field-id";
+  static final String ASSERT_CURRENT_SCHEMA_ID = "assert-current-schema-id";
+  static final String ASSERT_LAST_ASSIGNED_PARTITION_ID = "assert-last-assigned-partition-id";
+  static final String ASSERT_DEFAULT_SPEC_ID = "assert-default-spec-id";
+  static final String ASSERT_DEFAULT_SORT_ORDER_ID = "assert-default-write-order-id";
+
+  // AssertTableUUID
+  private static final String UUID = "uuid";
+
+  // AssertRefSnapshotID
+  // TODO - This is called `ref` in the spec.
+  //   https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1359-L1360
+  private static final String NAME = "name";
+  private static final String SNAPSHOT_ID = "snapshot-id";
+
+  // AssertLastAssignedFieldId
+  private static final String LAST_ASSIGNED_FIELD_ID = "last-assigned-field-id";
+
+  // AssertCurrentSchemaID
+  private static final String SCHEMA_ID = "current-schema-id";
+
+  // AssertLastAssignedPartitionId
+  private static final String LAST_ASSIGNED_PARTITION_ID = "last-assigned-partition-id";
+
+  // AssertDefaultSpecID
+  private static final String SPEC_ID = "default-spec-id";
+
+  // AssertDefaultSortOrderID
+  // TODO - Is currently referred to as default-write-order-id in spec but class and comments use sort-order. Need to
+  //  update it in class or in spec
+  private static final String SORT_ORDER_ID = "default-write-order-id";
```

Comment:
This doesn't necessarily need to be updated in the class or the spec, as we do use `sort-order-id` and `write-order-id` somewhat interchangeably.

I hesitated to update the class, because the associated `MetadataUpdate` class uses `order-id` in the spec, but the classes are all named `SortOrder`. https://github.com/apache/iceberg/blob/a19ac7835c8e251fd6a042e056f29ee63262adec/open-api/rest-catalog-open-api.yaml#L1231-L1240

Happy to update either one. We could also leave these two as not implemented, review the easier parts of this PR, and then add a second PR to handle it in isolation.

Reply:
Happy to update either one. We could also leave these two as not implemented, review the easier parts of this PR, and then add a second PR to handle it in isolation.

Reply:
I'd be fine renaming this to `assert-default-sort-order-id` and `default-sort-order-id` so that it matches `SetDefaultSortOrder` with `sort-order-id`. We'd need to update the spec.

Reply:
Ok. I'm somewhat partial to `write-order-id` as the condition is about how the data is written (and not how the data is necessarily read), but `sort-order-id` does match much more of the existing code.